src/util/types.ts

```diff
@@ -1050,13 +1050,35 @@ export interface GeometrySector {
 export type D3Scale<T> = D3ScaleContinuousNumeric<T, number>;
 
 export type AxisDomainItem = string | number | Function | 'auto' | 'dataMin' | 'dataMax';
-/** The domain of axis */
+/**
+ * The domain of axis.
+ *
+ * Can be specified in many ways:
+ * - array of numbers
+ * - with special strings like 'dataMin' and 'dataMax'
+ * - with special string math like 'dataMin - 100'
+ * - with keyword 'auto'
+ * - or a function
+ * - array of functions
+ * - or a combination of the above
+ *
+ * See https://recharts.org/en-US/api/XAxis#domain for examples
+ */
 export type AxisDomain =
   | string[]
   | number[]
   | [AxisDomainItem, AxisDomainItem]
   | (([dataMin, dataMax]: [number, number], allowDataOverflow: boolean) => [number, number]);
 
+/**
+ * SpecificDomain is an evaluated AxisDomain.
+ * Unlike AxisDomain, it has no variety - it's a tuple of two strings.
```

Comment:
Two strings? 🤔 The type below says number. 
Good point though, how about a categorical axis with strings on the xAxis? What does the domain look like in that case?

Reply:
Ah right I don't know what I was thinking.

Also yes I have discovered the existence of categorical axes after I wrote this comment :-X I'll update